src/components/calendar/CalendarList.tsx

```diff
@@ -0,0 +1,155 @@
+import { useTranslation } from 'language/LanguageProvider';
+import React, { useEffect, useState } from 'react';
+import { FlatList} from 'react-native';
+import { Divider, Title } from 'react-native-paper';
+
+import { Calendar, entryGt } from 'storage/calendar';
+import { CalendarListSection } from './CalendarListSection';
+
+/**
+ * Returns a new calendar grouped by date.
+ *
+ * @remarks
+ * This function assumes the calendar is sorted.
+ *
+ * @param calendar - The calendar to group by date (day)
+ * @returns A list of calendars where each calendar is a day of entries
+ */
+const groupByDate = (calendar: Calendar): Calendar[] => {
+  let groups: Calendar[] = [];
+  let currentGroup: Calendar = [];
+  let date = "";
+
+  calendar.forEach((entry) => {
+    if (entry.date == date) {
+      currentGroup.push(entry);
+    } else {
+      if (currentGroup.length !== 0){
+        groups.push(currentGroup);
+      }
+      currentGroup = [entry];
+      date = entry.date;
+    }
+  });
+
+  if (currentGroup.length !== 0) {
+    groups.push(currentGroup);
+  }
+
+  return groups;
+}
+
+/**
+ * Inserts headers of the form YYYY-mm before calendars that has the
+ * header as year and month.
+ *
+ * @remarks
+ * This function assumes the calendar is sorted.
+ *
+ * @param groups - The list of calendars to insert headers into
+ * @returns A list containing calendars and strings (headers)
+ */
+const insertMonthHeaders = (groups: Calendar[]): (Calendar | string)[] => {
+  let groupsWithHeaders: (Calendar | string)[] = [];
+  let currentMonth = "";
+
+  groups.forEach(group => {
+    const month = group[0].date.slice(0, 7); // YYYY-mm
+    if (month !== currentMonth) {
+      groupsWithHeaders.push(month);
+    }
+    groupsWithHeaders.push(group);
+    currentMonth = month;
+  });
+
+  return groupsWithHeaders;
+}
+
+/**
+ * Returns the current date in ISO format. Normal programming languages use
+ * something like date.strftime("%Y-%m-%d").
+ *
+ * @returns A ISO-formatted date
+ */
+const ISODate = (): string => {
+  const date = new Date();
+  const year = date.getFullYear();
+  let month = '' + (date.getMonth() + 1);
+  let day = '' + date.getDate();
+
+  if (month.length < 2) {
+    month = '0' + month;
+  }
+  if (day.length < 2) {
+    day = '0' + day;
+  }
+
+  return [year, month, day].join('-');
+}
+
+type ListState = {groups: Calendar[], entryCount: number};
+
+export const CalendarList: React.FC<{calendar: Calendar}> = ({ calendar }) => {
+  const [listState, setListState] = useState<ListState>({groups: [], entryCount: 0});
+  const dict = useTranslation();
+
```

Comment:
Yeah, I was wondering if we should put docs on components. Perhaps this is something we should discuss. I personally think it is not really necessary, though it might be good to know which props should be passed.

Reply:
On the other hand, the props are kind of obvious when you look at the function declaration.

Reply:
I wasn´t sure of what we said at the meeting and therefore ? in the end

Reply:
I would say that reusable components should have doc-strings but dedicated components should not have that requirement.

Reply:
I cannot think of any situation where these would need to be reused that is within our scope. I'm letting it be like this for now.